tests/codegen/enum/discriminant-not-immediate.rs

```diff
@@ -0,0 +1,28 @@
+// Checks that upon read/write of non-immediate enums that the discriminant has correct layout.
+//@ compile-flags: -C no-prepopulate-passes
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @write_option_to_memory(
+#[no_mangle]
+pub fn write_option_to_memory(store: &mut Option<u32>, reg: Option<u32>) {
```

Comment:
Here is an example of why we need `zeroext`: https://llvm.godbolt.org/z/5K1Ex16W1.

Please add a `CHECK-SAME` to check the parameter attribute.